src/System.Runtime.Serialization.Formatters/tests/FormatterServicesTests.cs

```diff
@@ -29,21 +34,260 @@ public void GetSerializableMembers_Interface()
+        public void GetUniniitalizedObject_SharedGenericInstance_NetCore_ThrowsNotSupportedException()
+        {
+            Type canonType = Type.GetType("System.__Canon");
+            Assert.NotNull(canonType);
+            Type sharedGenericInstance = typeof(GenericClass<>).MakeGenericType(canonType);
+            Assert.Throws<NotSupportedException>(() => FormatterServices.GetUninitializedObject(sharedGenericInstance));
+            Assert.Throws<NotSupportedException>(() => FormatterServices.GetSafeUninitializedObject(sharedGenericInstance));
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework, "The coreclr doesn't support GetUninitializedObject for shared generic instances")]
+        public void GetUniniitalizedObject_SharedGenericInstance_Netfx_InitializesValue()
+        {
+            Type canonType = Type.GetType("System.__Canon");
+            Assert.NotNull(canonType);
+            Type sharedGenericInstance = typeof(GenericClass<>).MakeGenericType(canonType);
+            Assert.Throws<NotSupportedException>(() => FormatterServices.GetUninitializedObject(sharedGenericInstance));
+            Assert.Equal(0, ((IGenericClass)FormatterServices.GetSafeUninitializedObject(sharedGenericInstance)).Value);
+        }
+
+        [Fact]
+        public void GetUninitializedObject_NullableType_InitializesValue()
+        {
+            int? nullableUnsafe = (int?)FormatterServices.GetUninitializedObject(typeof(int?));
+            Assert.True(nullableUnsafe.HasValue);
+            Assert.Equal(0, nullableUnsafe.Value);
+
+            int? nullableSafe = (int?)FormatterServices.GetSafeUninitializedObject(typeof(int?));
+            Assert.True(nullableSafe.HasValue);
+            Assert.Equal(0, nullableSafe.Value);
+        }
+
+        [Theory]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "The full .NET framework doesn't support GetUninitializedObject for COM objects")]
+        public void GetUninitializedObject_ContextBoundObjectSubclass_NetCore_InitializesValue()
+        {
+            Assert.Equal(0, ((COMObject)FormatterServices.GetUninitializedObject(typeof(COMObject))).Value);
+            Assert.Equal(0, ((COMObject)FormatterServices.GetSafeUninitializedObject(typeof(COMObject))).Value);
+        }
+
+        [Theory]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework, "The coreclr supports GetUninitializedObject for COM objects")]
+        public void GetUninitializedObject_ContextBoundObjectSubclass_Netfx_ThrowsNotSupportedException()
+        {
+            Assert.Throws<NotSupportedException>(() => FormatterServices.GetUninitializedObject(typeof(COMObject)));
+            Assert.Throws<NotSupportedException>(() => FormatterServices.GetSafeUninitializedObject(typeof(COMObject)));
+        }
+
+        public class COMObject : ContextBoundObject
```

Comment:
Shows how little I know about the runtime :D.

I've renamed this, and also added  a test for an actual COM object